core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -51,18 +51,21 @@ class ControllerChannelManager(controllerContext: ControllerContext,
                                time: Time,
                                metrics: Metrics,
                                stateChangeLogger: StateChangeLogger,
+                               val clusterId: String,
                                threadNamePrefix: Option[String] = None) extends Logging with KafkaMetricsGroup {
   import ControllerChannelManager._
 
   protected val brokerStateInfo = new HashMap[Int, ControllerBrokerStateInfo]
+  protected val remoteControllerStateInfo = new HashMap[Int, ControllerBrokerStateInfo]
```

Comment:
There seems to be a lot of similar processing for a remote controller host and a local broker host. What do you think of reusing the existing brokerStateInfo map to maintain both types of hosts? To tell which host is remote, there can be a separate map/set to record the remote broker ids.

Reply:
Yup, good idea.